pkgs/stdenv/generic/default.nix

```diff
@@ -19,6 +19,7 @@ let lib = import ../../../lib; in lib.makeOverridable (
 , setupScript ? ./setup.sh
 
 , extraNativeBuildInputs ? []
+, extraNativeBuildInputsPostStrip ? [] # TODO: this is terrible
```

Comment:
Exactly what the comment says. Signing has a dependency on `codesign_allocate`, which is prefixed with the `targetPrefix`, so it needs some way to know the prefix. The other files in this list use the variables introduced in the bintools setup hook, for example `strip.sh` uses `$STRIP`.